src/commands/application/logs.rs

```diff
@@ -0,0 +1,256 @@
+) -> Result<bool, CliError> {
+    let auth_progress_bar = new_spinner_progress_bar();
+    auth_progress_bar.set_message("Checking if you're authenticated to Google Cloud...");
+
+    let gcloud_client = GCloudClient::new().await;
+
+    auth_progress_bar.finish_and_clear();
+
+    let application_progress_bar = new_spinner_progress_bar();
+    application_progress_bar.set_message("Fetching application details ... ");
+
+    // Calling API ...
+    let client = QueryClientBuilder::default()
+        .with_access_token(
+            context
+                .config
+                .auth
+                .ok_or(CliError::UnAuthenticated)?
+                .id_token,
+        )
+        .with_sub(context.state.sub)
+        .with_api_url(context.config.core.wukong_api_url)
+        .build()?;
+
+    let application_resp = client
+        .fetch_application_with_k8s_cluster(
+            &context.state.application.clone().unwrap(),
+            &namespace.to_string().to_lowercase(),
+            &version.to_string().to_lowercase(),
+        ) // SAFERY: the application is checked on the caller so it will always be Some(x) here
+        .await?
+        .data
+        .unwrap()
+        .application;
+
+    if let Some(application_data) = application_resp {
+        if let Some(cluster) = application_data.k8s_cluster {
+            let filter = generate_filter(
+                &cluster.cluster_name,
+                &cluster.k8s_namespace,
+                since,
+                until,
+                show_error_and_above,
+            )?;
+            let resource_names = vec![format!("projects/{}", cluster.google_project_id)];
+
+            application_progress_bar.finish_and_clear();
+
+            let progress_bar = new_spinner_progress_bar();
+            progress_bar.set_message("Fetching log entries ... ");
+
+            let log = gcloud_client
+                .get_log_entries(LogEntriesOptions {
+                    resource_names: Some(resource_names),
+                    page_size: Some(*limit),
+                    filter: Some(filter),
+                    ..Default::default()
+                })
+                .await?;
+
+            progress_bar.finish_and_clear();
+
+            eprintln!("{log}");
+        }
+    }
+
+    Ok(true)
+}
+
+static TIMESTAMP_DAY_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^\d+d$").unwrap());
+static TIMESTAMP_HOUR_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^\d+h$").unwrap());
+static TIMESTAMP_MINUTE_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^\d+m$").unwrap());
+
+fn generate_filter(
+    cluster_name: &str,
+    namespace_name: &str,
+    since: &Option<String>,
+    until: &Option<String>,
+    show_error_and_above: &bool,
+) -> Result<String, CliError> {
+    let mut filter = String::new();
+    filter.push_str(format!("resource.type=\"k8s_container\" AND resource.labels.cluster_name=\"{}\" AND resource.labels.namespace_name=\"{}\"", cluster_name, namespace_name).as_str());
```

Comment:
also, we should print out the filter in `TRACE` log level so that we can double confirm what was being sent to Google Logging API when using the `-vvvv` flag.

Reply:
ok, I'll add this

Reply:
Added